src/table/table.component.tsx

```diff
@@ -151,29 +180,30 @@ const Table = React.memo((props: TableProps): React.ReactElement => {
 
   const { columnOrder } = state;
 
-  const columnOrderUpdater = (result: any): string[] => {
-    const items = Array.from(visibleColumns);
-    const [reorderedItem] = items.splice(result.source.index, 1);
-    items.splice(result.destination.index, 0, reorderedItem);
-
-    return items.map((column: Column) => {
-      return column.Header?.toString() ?? '';
-    });
-  };
-
   const handleOnDragEnd = (result: DropResult): void => {
-    if (!result.destination) return;
-    setColumnOrder(columnOrderUpdater(result));
+    if (result.destination)
+      setColumnOrder(columnOrderUpdater(result, visibleColumns));
   };
 
   return (
     <div>
       {loadedData && totalDataCount > 0 ? (
         <div>
           <div>
-            <MuiTableContainer component={Paper}>
+            <MuiTableContainer
+              role="table-container"
+              component={Paper}
+              sx={{ maxHeight: 300, overflow: 'auto' }}
+            >
               <MuiTable {...getTableProps()}>
-                <MuiTableHead>
+                <MuiTableHead
+                  sx={{
+                    position: 'sticky',
+                    background: 'white', // theme colour later on
```

Comment:
Can we add a TODO to this, as otherwise this comment could be forgotten about, whereas we know to search for TODO